src/django_components/extensions/dependencies.py

```diff
@@ -0,0 +1,21 @@
+from django_components.dependencies import cache_component_css, cache_component_js
+from django_components.extension import (
+    ComponentExtension,
+    OnComponentClassCreatedContext,
+)
+
+
+class DependenciesExtension(ComponentExtension):
+    """
+    This extension adds a nested `Dependencies` class to each `Component`.
+
+    This extension is automatically added to all components.
+    """
+
+    name = "dependencies"
+
+    # Cache the component's JS and CSS scripts when the class is created, so that
+    # components' JS/CSS files are accessible even before having to render the component first.
+    def on_component_class_created(self, ctx: OnComponentClassCreatedContext) -> None:
+        cache_component_js(ctx.component_cls, force=True)
+        cache_component_css(ctx.component_cls, force=True)
```

Comment:
This is where the components' JS/CSS files are cached. This is used for the HTML fragments support, so that the JS/ CSS can be lazily accessed via endpoint by our JS dependency manager from within the browser.

Implemented as an extension, so that 1) this runs for all components, and 2) so that this runs after Django is set up.